### ISSUE TITLE
Warning when compiling `custom_temporary_allocation.cu`

### ISSUE THREAD

**Issue by szsdk** [thrust/thrust#1007]
I try to compile `examples/cuda/custom_temporary_allocation.cu` with cuda 10.1 and get an error like this:
```
/usr/local/cuda-10.1/bin/../targets/x86_64-linux/include/thrust/mr/allocator.h(106): warning: calling a __host__ function("thrust::mr::stateless_resource_allocator<int,  ::thrust::system::cuda::detail::cuda_memory_resource<& ::cudaMalloc, & ::cudaFree,  ::thrust::cuda_cub::pointer<void> > > ::stateless_resource_allocator") from a __host__ __device__ function("thrust::mr::stateless_resource_allocator<int,  ::thrust::system::cuda::detail::cuda_memory_resource<& ::cudaMalloc, & ::cudaFree,  ::thrust::cuda_cub::pointer<void> > > ::stateless_resource_allocator [subobject]") is not allowed
```

**Comment by allisonvacanti** [comment]
This is being triggered by [`custom_temporary_allocation.cu:160`](https://github.com/thrust/thrust/blob/master/examples/cuda/custom_temporary_allocation.cu#L155-L160):

```
  thrust::host_vector<int> h_input(num_elements);
  // ...
  thrust::cuda::vector<int> d_input = h_input;
```

The `d_input` line is where the problematic instantiation occurs, so it's somewhere under `cuda::vector`'s converting constructor that takes a different parameterization of `vector_base`. I've dug around for a while in the vector, allocator, and memory resource code and didn't see anything that would cause this warning.

**Comment by allisonvacanti** [comment]
This was fixed a few weeks ago.
